clients/java/src/main/java/io/zeebe/client/impl/OAuthCredentialsProvider.java

```diff
@@ -73,9 +73,15 @@ public void applyCredentials(Metadata headers) {
   @Override
   public boolean shouldRetryRequest(Throwable throwable) {
     try {
-      return throwable instanceof StatusRuntimeException
-          && ((StatusRuntimeException) throwable).getStatus() == Status.UNAUTHENTICATED
-          && refreshCredentials();
+      LOG.debug("Deciding whether to retry error: ", throwable);
```

Comment:
I'm not sure this is particularly useful. I'm guessing the really interesting part here is only if we actually retry, no? If we don't retry the error should bubble up. So imo I'd only log if we will retry.